include/zephyr/bluetooth/services/cts.h

```diff
@@ -0,0 +1,139 @@
+	/* time changed due to dst offset change */
+	BT_CTS_UPDATE_REASON_DAYLIGHT_SAVING = BIT(3),
+};
+
+/**
+ * @brief Current Time service data format, Please refer to
+ * specifications for more details
+ */
+struct bt_cts_time_format {
+	uint16_t year;
+	uint8_t mon;
+	uint8_t mday;
+	uint8_t hours;
+	uint8_t min;
+	uint8_t sec;
+	uint8_t wday;
+	uint8_t fractions256;
+	uint8_t reason;
+} __packed;
+
+/** @brief Current Time Service callback structure */
+struct bt_cts_cb {
+	/** @brief Current Time Service notifications changed
+	 *
+	 * @param enabled Flag that is true if notifications were enabled, false
+	 *                if they were disabled.
+	 */
+	void (*notification_changed)(bool enabled);
+
+	/**
+	 * @brief Current Time is updated using Gatt Attr write, It is the
+	 * responsibility of the application to save this information locally
+	 * and use it for future usage
+	 *
+	 * @param cts_time [IN] updated time information received
+	 *
+	 * @return 0 application has decoded it successfully
+	 *           negative error codes on failure
+	 *
+	 */
+	int (*cts_time_write)(struct bt_cts_time_format *cts_time);
+
+	/**
+	 * @brief When current time Read request or notification is triggered, CTS uses
+	 * this callback to retrieve current time information from application. Application
+	 * must implement it and provide cts formatted current time information
+	 *
+	 * @param cts_time [IN]
+	 *
+	 * @return 0 application has encode it successfully
+	 *           negative error codes on failure
+	 */
+	int (*fill_current_cts_time)(struct bt_cts_time_format *cts_time);
+};
+
+/**
+ * @brief This API should be called at boot to ensure proper functioning of CTS service
+ *
+ * @param cb pointer to required callback
+ * @return int   0 on success
+ *               negative error codes on failure
+ */
+int bt_cts_init(const struct bt_cts_cb *cb);
+
+/**
+ * @brief Update CTS Server time and notify all clients
+ * that have notifications enabled
+ *
+ * @param reason update reason to be sent to the clients
+ * @return int   0 on success
+ *               negative error codes on failure
+ */
+int bt_cts_send_notification(enum bt_cts_update_reason reason);
+
+/**
+ * @brief Helper API to decode CTS formatted time into milliseconds since epoch
+ *
+ * @param ct_time [IN]  cts time formatted time
+ * @param unix_ms [OUT] pointer to store parsed millisecond since epoch
+ * @return int    0 on success
+ *                negative error codes on failure
+ */
+int bt_cts_time_to_unix_ms(const struct bt_cts_time_format *ct_time, int64_t *unix_ms);
+
+/**
+ * @brief Helper API to encode milliseconds since epoch to CTS formatted time
+ *
+ * @param ct_time [OUT] Pointer to store CTS formatted time
+ * @param unix_ms [IN]  milliseconds since epoch to be converted
+ * @return int    0 on success
+ *                negative error codes on failure
+ */
+int bt_cts_time_from_unix_ms(struct bt_cts_time_format *ct_time, int64_t unix_ms);
```

Comment:
Might still make sense to mention which Kconfig option these depend on. There's going to be a convenience helper available for it soon, but it hasn't been merged yet: #76783

Reply:
followed settings load reference, but now realised that check is for complete subsystem.
updated comment using reference from `sys_poweroff` API